src/Symfony/Component/Lock/Store/ZookeeperStore.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockReleasingException;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+use Zookeeper;
+use Throwable;
+
+/**
+ * ZookeeperStore is a StoreInterface implementation using Zookeeper as store engine.
+ *
+ * @author Ganesh Chandrasekaran <gchandrasekaran@wayfair.com>
+ */
+class ZookeeperStore implements StoreInterface
```

Comment:
More of an issue that already exists, but shouldn't this be outside of the core? Possibly a bridge. The problem already exists though, because of redis and memcached. However, adding multiple implementation in the core, might lead to the same issue as monolog has with all the different processors.

This class also has a dependency on Zookeeper, which I'm assuming is either a package or extension, which should be a dependency in the composer.json (hence it shouldn't be in the core).

Reply:
You are correct, a Zookeeper extension will be required. Can you create a new issue for this and can we leave that decision out of this PR's scope? Happy to help when you have the new issue for this.

Reply:
I think it falls under the scope of this PR, because we'd be introducing new code with(out) a dependency. Test should reflect this.

Ideally I'd see this in a ZookeeperBridge (standalone package), only this factory is not compatible with the proposed setup. I do think this factory is plain wrong by design as it's not capable of being plug&play. Perhaps @jderusse can comment on this part design wise?

Reply:
@iltar @jderusse - I have moved the new Data Store out of `Component` to `Bridge` now.

Reply:
Leaving it as is as other core members prefer to have this here.